substratevm/src/com.oracle.svm.configure/src/com/oracle/svm/configure/trace/AccessAdvisor.java

```diff
@@ -33,6 +35,12 @@
  * {@code AccessVerifier} classes which accesses to ignore when the agent is in restriction mode.
  */
 public final class AccessAdvisor {
+    /**
+     * {@link java.lang.reflect.Proxy} generated classes can be put in arbitrary packages depending
+     * on the visibility and module of the interfaces they implement, so we can only match against
+     * their class name using this pattern (which we hope isn't used by anything else).
+     */
+    public static final Pattern PROXY_CLASS_NAME_PATTERN = Pattern.compile("^(.+[/.])?\\$Proxy[0-9]+$");
```

Comment:
This will definitely result in false positives. What is the worst that can happen for a false positive?

Reply:
I don't think that it's particularly prone to false positives: the class name has to start with `$` and end with a sequence of digits so this does not match `Proxy` or `Proxy1` or inner classes like `Connection$Proxy1`. I can imagine that other class generators could use a similar naming scheme though. The worst that can happen is that these classes are omitted from the resulting agent configurations.